core/thread.c

```diff
@@ -104,6 +105,26 @@ void thread_yield(void)
     thread_yield_higher();
 }
 
+void thread_add_to_list(list_node_t *list, thread_t *thread)
+{
+    assert (thread->status < STATUS_ON_RUNQUEUE);
+
+    uint16_t my_prio = thread->priority;
+    list_node_t *new_node = (list_node_t*)&thread->rq_entry;
+    new_node->next = NULL;
+
+    while (list->next) {
+        thread_t *list_entry = container_of((clist_node_t*)list->next, thread_t, rq_entry);
+        if (list_entry->priority > my_prio) {
+            new_node->next = list->next;
+            break;
+        }
+        list = list->next;
+    }
+
+    list->next = new_node;
```

Comment:
ups, overlooked l119, so everything should be ok. You might want to more l119 out of the loop and remove l114 though.